main.go

```diff
@@ -40,22 +40,30 @@ func main() {
 		fmt.Printf("%s\n", config.GetDefaultConfig())
 		os.Exit(0)
 	}
+
 	appConfig, err := config.NewAppConfig("lazygit", version, commit, date, buildSource, *debuggingFlag)
 	if err != nil {
 		log.Fatal(err.Error())
 	}
 
-	app, err := app.NewApp(appConfig)
-
-	if err == nil {
-		err = app.Run()
+	appDir := flag.Arg(0)
+	if appDir != "" {
+		err := os.Chdir(appDir)
+		if err != nil {
+			log.Fatalf("Could not change directory to %s", appDir)
+		}
 	}
 
+	app, err := app.NewApp(appConfig)
 	if err != nil {
 		newErr := errors.Wrap(err, 0)
 		stackTrace := newErr.ErrorStack()
 		app.Log.Error(stackTrace)
 
-		log.Fatal(fmt.Sprintf("%s\n\n%s", app.Tr.SLocalize("ErrorOccurred"), stackTrace))
+		log.Fatalf("%s\n\n%s", app.Tr.SLocalize("ErrorOccurred"), stackTrace)
+	}
+
+	if err := app.Run(); err != nil {
+		log.Fatal(err)
```

Comment:
The code above may have looked a little weird but the intention was to have the stack trace shown if we error either when calling NewApp or calling app.Run(). So I think either we should pull the code in lines 59-63 into its own method and then call that with the error in both cases, ore move app.Run() back to where it was

Reply:
I think this would do the trick then. Whats your opinion?
```
app, err := app.NewApp(appConfig)
if err == nil {
    err = app.Run()
}

if err != nil {
  ...
}

Reply:
yep that looks good